File: Easygym.Domain/Entities/WorkoutSession.cs
using System.ComponentModel.DataAnnotations;

namespace Easygym.Domain.Entities
{
    public class WorkoutSession
    {
        public int Id { get; set; }
        public required int WorkoutId { get; set; }
        public Workout? Workout { get; set; }
        public required int TraineeId { get; set; }
        public User? Trainee { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        [Range(1, 10)]
        public int? PerceivedDifficulty { get; set; }
        public string? Notes { get; set; }
    }
}

File: Client/src/types/WorkoutSession.ts
import { User } from "@/types/User";
import { Workout } from "@/types/Workout";

export interface WorkoutSession {
    id: number;
    workoutId: number;
    workout?: Workout;
    traineeId: number;
    trainee?: User;
    startTime: string;
    endTime: string;
    perceivedDifficulty?: number;
    notes?: string;
  }

export interface CreateWorkoutSessionRequest {
    workoutId: number;
    traineeId: number;
    startTime: string;
    endTime: string;
    perceivedDifficulty?: number;
    notes?: string;
}

export interface UpdateWorkoutSessionRequest {
    perceivedDifficulty?: number;
    notes?: string;
}